ta/pkcs11/src/pkcs11_helpers.c

```diff
@@ -161,6 +161,32 @@ static const struct any_id __maybe_unused string_rc[] = {
 	PKCS11_ID(PKCS11_RV_NOT_IMPLEMENTED),
 };
 
+/*
+ * Conversion between PKCS11 TA and GPD TEE return codes
+ */
+enum pkcs11_rc tee2pkcs_error(TEE_Result res)
+{
+	switch (res) {
+	case TEE_SUCCESS:
+		return PKCS11_CKR_OK;
+
+	case TEE_ERROR_BAD_PARAMETERS:
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	case TEE_ERROR_OUT_OF_MEMORY:
+		return PKCS11_CKR_DEVICE_MEMORY;
+
+	case TEE_ERROR_SHORT_BUFFER:
+		return PKCS11_CKR_BUFFER_TOO_SMALL;
+
+	case TEE_ERROR_MAC_INVALID:
```

Comment:
for info: `TEE_ERROR_SIGNATURE_INVALID` from `TEE_AsymmetricVerifyDigest()` also matches `PKCS11_CKR_SIGNATURE_INVALID`.

Reply:
OK, I'll add that here.